### PR TITLE
Report call when connected

### PR DESCRIPTION
**Pull request by bobiechen-twilio**
Use the `reportOutgoingCallWithUUID:connectedAtDate:` method to notify the (CallKit) system that the call is now connected otherwise the system will treat it as a canceled call even if the call was accepted and connected.

### CODE REVIEW

**Review by ceaglest — COMMENTED** [review]
I believe this does solve the issue of failed calls, but a more comprehensive CallKit implementation would handle the `CXStartCallAction` by:

1. Reporting the outgoing call when it starts connecting.
2. Not fulfilling the `CXStartCallAction` until `callDidConnect:` is raised.
3. Failing the action if `call:didFailWithError:` is called.

An example (using block based syntax for connecting) can be found in [video-quickstart-swift](https://github.com/twilio/video-quickstart-swift/blob/master/VideoCallKitQuickStart/ViewController%2BCallKit.swift#L41).

Similarly when performing a `CXAnswerCallAction`, a well behaved implementation should:

1. Fail the action if accepting the invite fails synchronously.
2. Fulfill the `CXAnswerCallAction` when `callDidConnect:` is raised.
3. Failing the action if `call:didFailWithError:` is called.

Here is the answer implementation from [video-quickstart-swift](https://github.com/twilio/video-quickstart-swift/blob/master/VideoCallKitQuickStart/ViewController%2BCallKit.swift#L62). I'm not a CallKit expert but based on experience with Twilio Video I believe this is the preferred way to go.

The question to address is should we make these changes in this PR, or as a follow on? I'm okay with either.

File: ObjCVoiceCallKitQuickstart/ViewController.m
Comment (ceaglest) [inline ObjCVoiceCallKitQuickstart/ViewController.m:185]: Minor - you could also nil `self.callKitCompletionCallback` after calling it here, and in the failure case.

File: ObjCVoiceCallKitQuickstart/ViewController.m
Comment (ceaglest) [inline ObjCVoiceCallKitQuickstart/ViewController.m:303]: You should also fulfill the action here.

File: ObjCVoiceCallKitQuickstart/ViewController.m
Comment (ceaglest) [inline ObjCVoiceCallKitQuickstart/ViewController.m]: Fulfill the action upon success.

**Review by ceaglest — CHANGES_REQUESTED** [review]
Looks great! I believe there is one last piece of feedback to address before getting this merged. Since I'm out today, please get a final approval from @ptankTwilio.

File: ObjCVoiceCallKitQuickstart/ViewController.m
Comment (piyushtank) [inline ObjCVoiceCallKitQuickstart/ViewController.m]: 👍

File: ObjCVoiceCallKitQuickstart/ViewController.m
Comment (piyushtank) [inline ObjCVoiceCallKitQuickstart/ViewController.m]: 👍

**Review by piyushtank — APPROVED** [review]
LGTM!

File: ObjCVoiceCallKitQuickstart/ViewController.m
Comment (fabriziomoscon) [inline ObjCVoiceCallKitQuickstart/ViewController.m:327]: @bchen-twilio I was just wondering whether this last `[action fulfill]` is correct or it is left from the refactor. Because I see that is is called inside the success completion block.

Comment (bobiechen-twilio) [reply ObjCVoiceCallKitQuickstart/ViewController.m:327]: Hi @fabriziomoscon 

For some reason, I had to keep `[action fulfill]` in the `provider:performAnswerCallAction:` method otherwise the Voice iOS SDK won't be able to initialize & start the call audio properly.

-bobie

**Pull request merged**
